distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso

```diff
@@ -681,100 +650,103 @@ type Table
+           and an empty result is reported.
+         - If a column index is invalid, an `Index_Out_Of_Bounds` is
+           reported and an empty result is reported.
+         - If a join condition correlates columns whose types are not compatible
+           (for example comparing numeric types with text), an
+           `Invalid_Value_Type` is reported.
+         - If decimal columns are joined on equality, a
+           `Floating_Point_Grouping` is reported.
+
+         In any of the above cases, if a problem occurs, the resulting table
+         will have the desired structure, but it will be empty to indicate that
+         the join has failed due to an erroneous join condition.
+
+       ? Column Renaming
+
+         If columns from the two tables have colliding names, a prefix (by
+         default `Right_`) is added to the name of the column from the right
+         table. The left column remains unchanged. It is possible that the new
+         name will be in use, in this case it will be resolved using the normal
+         renaming strategy - adding subsequent `_1`, `_2` etc.
+
+       ? Result Ordering
+
+         The ordering of rows in the resulting table is not specified.
+
+       ? Joining on equality of columns with the same name
+
+         When joining two columns with the same name and an equality condition,
+         only one copy of column will be included in the output (avoiding
+         unnecessary duplication and renaming).
+
+       ? Same-name column join shorthand
+
+         As a shorthand, providing a column name or a list of column names
+         allows to join the two tables on equality of corresponding columns with
+         the same name. So `table.join other on=["A", "B"]` is a shorthand for:
+             table.join other on=[Join_Condition.Equals "A" "A", Join_Condition.Equals "B" "B"]
+    join : Table -> Join_Kind -> Join_Condition | Text | Vector (Join_Condition | Text) -> Text -> Problem_Behavior -> Table
+    join self right join_kind=Join_Kind.Inner on=[Join_Condition.Equals 0 0] right_prefix="Right_" on_problems=Report_Warning =
+        can_proceed = if Table_Helpers.is_table right . not then Error.throw (Type_Error.Error Table right "right") else
+            same_backend = case right of
+                            _ : Table -> True
+                            _ -> False
+            if same_backend . not then Error.throw (Illegal_Argument.Error "Currently cross-backend joins are not supported. You need to upload the table in-memory table before joining it with a database one, or materialize this table.") else
+                True
+        if can_proceed then
+            left = self
+            new_table_name = left.name + "_" + right.name
+
+            needed_indicators = case join_kind of
+                Join_Kind.Left_Exclusive  -> Pair.new False True
+                Join_Kind.Right_Exclusive -> Pair.new True False
+                _ -> Pair.new False False
```

Comment:
```suggestion
            needed_indicators = Pair.new (join_kind == Join_Kind.Left_Exclusive) (join_kind == Join_Kind.Right_Exclusive)
```
maybe that looks simpler?

Reply:
But this new code is wrong - it's actually reverse - `Left_Exclusive` join needs an indicator for _right_ column.

My opinion is that I prefer the original approach - it explicitly shows the control flow - for this kind of join I select that, for this that etc. Here it is a bit less explicit.

If anything, I'd do
```suggestion
            left_indicator_needed = join_kind == Join_Kind.Right_Exclusive
            right_indicator_needed = join_kind == Join_Kind.Left_Exclusive
```

I guess that could work OK if you think this is better. I still kind-of prefer the original one for showing the various branches more directly, but I admit both have pros and cons.